test/e2e/storage/csi_volumes.go

```diff
@@ -151,6 +158,32 @@ var _ = utils.SIGDescribe("CSI Volumes", func() {
 					claim.Spec.StorageClassName = &scName
 				}
 				testsuites.TestDynamicProvisioning(t, cs, claim, class)
+
+				By("Updating the driver")
+				ds, err = cs.AppsV1().DaemonSets(f.Namespace.Name).Get(ds.Name, metav1.GetOptions{})
+				framework.ExpectNoError(err, "Failed to get DaemonSet: %v", err)
+
+				expectedUpdatedPods := ds.Status.DesiredNumberScheduled
+				framework.Logf("DaemonSet with driver has %d scheduled pods", expectedUpdatedPods)
+				// For debugging:
+				framework.Logf("DaemonSet status: %+v", ds.Status)
+
+				framework.UpdateDaemonSetWithRetries(cs, f.Namespace.Name, ds.Name, func(ds *appsv1.DaemonSet) {
+					// Simulate driver update by adding a new label in DaemonSet template. This issues rolling update.
+					if ds.Spec.Template.Labels == nil {
+						ds.Spec.Template.Labels = map[string]string{}
+					}
+					ds.Spec.Template.Labels[f.UniqueName] = ""
+				})
+				framework.ExpectNoError(err, "Failed to update DaemonSet: %v", err)
+
+				By("Waiting for the update to complete")
+				selector := labels.SelectorFromSet(labels.Set(map[string]string{f.UniqueName: ""}))
```

Comment:
not sure what this selector is for, seems to be the nil selector. It is not clear to me how this is getting the Daemonset pods.

Reply:
see some lines above https://github.com/kubernetes/kubernetes/pull/70578/files/c0770267040509587d7ba503eee2d6e55cce95cf#diff-0d9ecaa3e6a0297186ad33f57aad472eR176

Reply:
Yeah, adding unique label a few lines above triggers rolling update of the DaemonSet, which simulates driver update. Waiting for pods with the label then basically waits until the update is done and new pod(s) are running.